mod.d/speculativeexecution.yaml

```diff
@@ -0,0 +1,196 @@
+              elif re.search("AuthenticAMD", cpuinfo_file.read()):
+                  print("Running on an AMD CPU.\n")
+                  return False
+              else:
+                  print("Running on a non-Intel or AMD CPU. This is currently unsupported.\n")
+                  sys.exit(0)
+      except Exception:
+          print("[WARN] Unable to determine CPU vendor\n")
+          raise
+
+
+  def get_kpti_status():
+      """
+      Determines if KPTI is in use to mitigate Meltdown.
+
+      Returns:
+         True (bool): If KPTI is enabled
+      """
+      try:
+          with open("/sys/devices/system/cpu/vulnerabilities/meltdown", "r") as meltdown_file:
+              meltdown = meltdown_file.read()
+              print(meltdown)
+              if re.search("PTI", meltdown):
+                  print("Kernel Page Table Isolation is enabled as a Meltdown mitigation\n")
+                  return True
+              else:
+                  print("Kernel Page Table Isolation is disabled as a Meltdown mitigation\n")
+                  with open("/proc/cmdline", "r") as cmdline_file:
+                      if re.search("nopti", cmdline_file.read()):
+                          print("Kernel Page Table isolation is disabled via nopti on the kernel commandline.\n")
+                  return False
+      except Exception:
+          print("[WARN] Unable to determine Meltdown mitigation status\n")
+          raise
+
+  def get_retpoline_status():
+      """
+      Determines if retpoline is in use to mitigate Spectre.
+
+      Returns:
+         True (bool): If retpoline is in use in the kernel
+      """
+      try:
+          with open("/sys/devices/system/cpu/vulnerabilities/spectre_v2", "r") as spectrev2_file:
+              spectrev2 = spectrev2_file.read()
+              print(spectrev2)
+              if re.search("retpoline", spectrev2):
+                  print("Retpoline is in use to mitigate Spectre V2\n")
+                  return True
+              elif re.search("Vulnerable: Retpoline", spectrev2):
+                      print("Retpoline is in use to mitigate Spectre V2, but IBPB is not showing as enabled.\n")
+                      return True
+              else:
+                  print("Retpoline is not in use to mitigate Spectre V2\n")
+                  return False
+      except Exception:
+          print("[WARN] Unable to determine Spectre V2 mitigation status\n")
+          raise
+
+  def get_pcid_status():
+      """
+      Determine if the system supports pcid or nvpcid. Pre-Haswell CPUs should only have PCID support.
+
+      Returns:
+          pcid (string): Highest PCID implementation supported
+      """
+      try:
+          with open("/proc/cpuinfo", "r") as cpuinfo_file:
+              if re.search("invpcid", cpuinfo_file.read()):
+                  print("invpcid supported. This is the best case for Kernel Page Table Isolation performance.\n")
+                  pcid = "invpcid"
```

Comment:
You could just return the string here since the pcid variable isn't used. This also applies to the elif block right below this if block.

Reply:
Can do.